core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -444,6 +444,12 @@ private void validateNoNewDeletesForDataFiles(
     if (parent == null || base.formatVersion() < 2) {
       return;
     }
+    List<Snapshot> snapshots =
+        Lists.newArrayList(
+            SnapshotUtil.ancestorsBetween(
+                base.currentSnapshot().snapshotId(), startingSnapshotId, base::snapshot));
+    boolean ignorePositionDeletes =
+        snapshots.stream().allMatch(this::cannotContainPosDeletesForPreviousSnapshots);
```

Comment:
I think `hasNoConflictingPosDeletes` might  be more appropriate, because this step is to judge whether the pos deletes in the new snapshots cannot have conflicts.